### DISCUSSION TITLE
How do I apply an Avalonia theme only to a specific control?

### DISCUSSION THREAD

**Discussion by V0du** [AvaloniaUI/Avalonia#15014] [Q&A] [answered]
My application uses control themes from another library (applied in App.axaml Application.Styles)
I want to apply Avalonia's default fluent theme to a specific ProgressBar contained in a datagrid cell.
Do I have to set the Theme property of the progress bar to a specific resource key? If so what is it?

```
<DataGridTemplateColumn.CellTemplate>
  <DataTemplate x:DataType="vm:MyCustomViewModel">
    <ProgressBar Theme="{StaticResource ???}" Minimum="0" Maximum="100" Value="{Binding PValue}" ShowProgressText="True" />
  </DataTemplate>
</DataGridTemplateColumn.CellTemplate>
```

**Comment by timunie** [comment]
You can set `RequestedThemeVariant` on any child node afaik

**Reply by thevortexcloud** [reply]
Wont that just set dark/light/whatever other variant rather than use an entirely different theme?

**Reply by timunie** [reply]
Yeah that's true

**Comment by Oudoum** [comment]
You can access the "default" theme by using the ControlTheme x:Key of the Element.
Just open up the ProgressBar.xaml and you will find **{x:Type ProgressBar}**.


`<DataGridTemplateColumn.CellTemplate> <DataTemplate x:DataType="vm:MyCustomViewModel"> <ProgressBar Theme="{StaticResource {x:Type ProgressBar}}" Minimum="0" Maximum="100" Value="{Binding PValue}" ShowProgressText="True" /> </DataTemplate> </DataGridTemplateColumn.CellTemplate>`

**Reply by V0du** [reply]
I've tried this and it doesn't seem to work, the progressbar still has the other library's theme applied.

**Reply by Oudoum** [reply, marked as answer]
Well this is what I expected.
Then you have to manually create a new Style or Theme and change it back to the default property values.
Or copy paste the whole ControlTheme and change the x:Key.
